src/components/views/rooms/RoomHeader.tsx

```diff
@@ -19,16 +19,31 @@ import React from "react";
 import type { Room } from "matrix-js-sdk/src/models/room";
 import { IOOBData } from "../../../stores/ThreepidInviteStore";
 import { useRoomName } from "../../../hooks/useRoomName";
+import DecoratedRoomAvatar from "../avatars/DecoratedRoomAvatar";
+import { RightPanelPhases } from "../../../stores/right-panel/RightPanelStorePhases";
+import RightPanelStore from "../../../stores/right-panel/RightPanelStore";
+import { useTopic } from "../../../hooks/room/useTopic";
 
 export default function RoomHeader({ room, oobData }: { room?: Room; oobData?: IOOBData }): JSX.Element {
     const roomName = useRoomName(room, oobData);
+    const roomTopic = useTopic(room);
 
     return (
-        <header className="mx_RoomHeader light-panel">
-            <div className="mx_RoomHeader_wrapper">
-                <div className="mx_RoomHeader_name" dir="auto" title={roomName} role="heading" aria-level={1}>
+        <header
+            className="mx_RoomHeader light-panel"
+            onClick={() => {
+                const rightPanel = RightPanelStore.instance;
+                rightPanel.isOpen
+                    ? rightPanel.togglePanel(null)
+                    : rightPanel.setCard({ phase: RightPanelPhases.RoomSummary });
+            }}
+        >
+            {room && <DecoratedRoomAvatar room={room} oobData={oobData} avatarSize={40} displayBadge={false} />}
```

Comment:
Switch between the two components depending on whether or not you have a room or make DecoratedRoomAvatar's room prop optional like RoomAvatar? The whole point of `oobData.avatarUrl` is to be able to render an avatar even if you don't have a room object

Reply:
Either that or make `room` required like LegacyRoomHeader

Reply:
So this behaviour is currently not possible. I've added a task to deal with that https://github.com/vector-im/element-web/issues/25902 but this is probably not the right pull request to deal with it.

Reply:
```suggestion
            {room ? <DecoratedRoomAvatar room={room} oobData={oobData} avatarSize={40} displayBadge={false} /> : <RoomAvatar oobData={oobData} width={40} height={40} />}
```

Reply:
I don't think we should embed the logic here. But it is done so that we can move on and merge this pull request.

`Room` is marked as optional because it is the reality of the types. `LegacyRoomHeader` types are wrong. 
The room object is passed from the `RoomContext` and the `room` property is `Room | undefined` there. I have not spent the time investigating nor decided how i'm going to fix it properly moving forward.

Reply:
The primary call site of LegacyRoomHeader is in RoomView, this is guarded by this block which ensures that a falsey room is not passed

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/002783f2-e9a7-4328-bab7-e5c4cab77dcf)

Typescript reflects this type narrowing

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/3ec49a6e-19d8-4b3a-92bb-cfbc25e3fa23)

The call site in WaitingForThirdPartyRoomView is similarly guarded though indirectly
![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/7e006951-118b-4356-8b2d-c8214382fb65)

The one in LocalRoomView is type asserted but then `room` is accessed so that would throw before RoomHeader, again causing narrowing

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/139c6aa7-efe4-4e6a-9f91-e2ab47690758)

The only one breaking assertions is LocalRoomCreateLoader

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/9c4a7720-c009-4858-9984-5785ade6b4ed)

Reply:
Sounds good. I've pushed your suggestion, so we should probably keep it like this for this pull request to not have other matters creep in.

Unless you have other suggestions, we should be good to go?